pkg/controller/certificate-shim/checks.go

```diff
@@ -19,28 +19,34 @@ package controller
 import (
 	"fmt"
 
-	networkingv1beta1 "k8s.io/api/networking/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	v1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 )
 
-func (c *controller) ingressesForCertificate(crt *v1.Certificate) ([]*networkingv1beta1.Ingress, error) {
-	ings, err := c.ingressLister.List(labels.NewSelector())
+// ownersOfCertificate returns references to controllers that are referred to in the
+// given Certificate's ownerRef.
+func (c *controller) ownersOfCertificate(crt *v1.Certificate) ([]runtime.Object, error) {
+	objs, err := c.objectLister.List(labels.NewSelector())
 
 	if err != nil {
 		return nil, fmt.Errorf("error listing certificates: %s", err.Error())
 	}
 
-	var affected []*networkingv1beta1.Ingress
-	for _, ing := range ings {
-		if crt.Namespace != ing.Namespace {
+	var affected []runtime.Object
+	for _, obj := range objs {
+		o, ok := obj.(metav1.Object)
+		if !ok {
+			panic(fmt.Sprintf("object %T is not a metav1.object", obj))
```

Comment:
Do we have to panic here? I realize this can never really happen, but isn't great in the code.

Can we return error instead?